actors/evm/tests/contracts/NotificationReceiver.sol

```diff
@@ -0,0 +1,523 @@
+    }
+    
+    SectorNotification[] public notifications;
+    mapping(uint64 => uint256[]) public sectorNotificationIndices;
+    
+    // Counter for total notifications received
+    uint256 public totalNotifications;
+    
+    // Flag to test different response behaviors
+    bool public shouldRejectNotifications = false;
+    
+    // Method selector for handle_filecoin_method
+    bytes4 constant NATIVE_METHOD_SELECTOR = 0x868e10c4;
+    
+    // Sector content changed method number
+    uint64 constant SECTOR_CONTENT_CHANGED = 2034386435;
+    
+    /**
+     * @dev Get the count of notifications for a specific sector
+     */
+    function getNotificationCount(uint64 sector) public view returns (uint256) {
+        return sectorNotificationIndices[sector].length;
+    }
+    
+    /**
+     * @dev Get all notification indices for a sector
+     */
+    function getSectorNotifications(uint64 sector) public view returns (uint256[] memory) {
+        return sectorNotificationIndices[sector];
+    }
+    
+    /**
+     * @dev Get a specific notification by index
+     */
+    function getNotification(uint256 index) public view returns (
+        uint64 sector,
+        int64 minimumCommitmentEpoch,
+        bytes memory dataCid,
+        uint64 pieceSize,
+        bytes memory payload
+    ) {
+        require(index < notifications.length, "Invalid notification index");
+        SectorNotification memory notif = notifications[index];
+        return (
+            notif.sector,
+            notif.minimumCommitmentEpoch,
+            notif.dataCid,
+            notif.pieceSize,
+            notif.payload
+        );
+    }
+    
+    /**
+     * @dev Handle incoming Filecoin method calls
+     * This is the main entry point for receiving notifications from the miner actor
+     */
+    function handle_filecoin_method(uint64 method, uint64 inCodec, bytes memory params) public returns (uint64, uint64,bytes memory) {
+        // 0x51 is IPLD CBOR codec 
+        require(inCodec == 0x51, "Invalid codec");
+        // Check if this is a sector content changed notification
+        if (method == SECTOR_CONTENT_CHANGED) {
+            bytes memory ret = processSectorContentChanged(params);
+            uint64 codec = 0x51;
+            return (0, codec, ret);
+        }
+        
+        // For other methods, just revert
+       revert("Invalid method");
+    }
+    
+    /**
+     * @dev Process sector content changed notification
+     * Expected params structure (CBOR encoded):
+     * {
+     *   sectors: [{
+     *     sector: uint64,
+     *     minimum_commitment_epoch: int64,
+     *     added: [{
+     *       data: Cid,
+     *       size: uint64,
+     *       payload: bytes
+     *     }]
+     *   }]
+     * }
+     *
+     * All notifications are accepted so CBOR true returned for every piece of every notified sector
+     */
+    function processSectorContentChanged(bytes memory params) internal returns (bytes memory) {
```

Comment:
A nice thing about this test is that this is a little reference implementation for how to handle DDO notifications